scaleway/resource_baremetal_server.go

```diff
@@ -272,6 +309,58 @@ func resourceScalewayBaremetalServerUpdate(ctx context.Context, d *schema.Resour
 		return diag.FromErr(err)
 	}
 
+	server, err := baremetalAPI.GetServer(&baremetal.GetServerRequest{
+		Zone:     zonedID.Zone,
+		ServerID: zonedID.ID,
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var serverGetOptionIDs []string
+	for i := range server.Options {
+		serverGetOptionIDs = append(serverGetOptionIDs, server.Options[i].ID)
+	}
+
+	if d.HasChange("option_ids") {
+		optionIDs := expandStrings(d.Get("option_ids").(*schema.Set).List())
+		if len(optionIDs) > 0 {
+			optionsToAdd := baremetalCompareOptionIDsToAdd(optionIDs, serverGetOptionIDs, zonedID.Zone)
+			for i := range optionsToAdd {
+				_, err = baremetalAPI.AddOptionServer(&baremetal.AddOptionServerRequest{
```

Comment:
We need to check the [state](https://github.com/scaleway/scaleway-sdk-go/blob/master/api/baremetal/v1/baremetal_sdk.go#L414)

Reply:
Added a waitForServerOptions to check the state